lib/src/platform_download.dart

```diff
@@ -0,0 +1,137 @@
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:crypto/crypto.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_downloader/flutter_downloader.dart';
+import 'package:flutter_downloader/src/desktop_platform_download.dart';
+import 'package:flutter_downloader/src/download_metadata.dart';
+
+/// A platform specific [Download] which invokes the method channel and make the
+/// download native if the platform has this requirement.
+class PlatformDownload extends DesktopPlatformDownload {
```

Comment:
I can use `PlatformDownload` on mobile, yet it inherits from `DesktopPlatformDownload`? Seems a bit wrong to me.